SearchEngine/lambda_SearchEngine/include/Server_EchoServer.h

```diff
@@ -0,0 +1,145 @@
+#ifndef __ECHOSERVER_H__
+#define __ECHOSERVER_H__
+
+#include <sw/redis++/redis++.h>
+#include <unistd.h>
+
+#include <iostream>
+#include "Server_Thread.h"
+#include "CppJieba.h"
+#include "KeyRecommander.h"
+#include "Server_TcpConnection.h"
+#include "Server_TcpServer.h"
+#include "Server_ThreadPool.h"
+#include "WebPageSearcher.h"
+#include "CacheManager.h"
+#include "log4pp.h"
+
+using std::cout;
+using std::endl;
+using namespace sw::redis;
+
+
+class MyTask {
+public:
+    MyTask(const string &msg, const TcpConnectionPtr &con,
+           CnDictionary &cndict, EngDictionary &engdict, sw::redis::Redis &redis)
+        : _msg(msg), _con(con), _cndcit(cndict), _engdict(engdict), _redis(redis) {
+        }
+
+    void process() {
+        // 切出msg的前1位int,设为flag
+        auto & _currCache = CacheManager::GetInstance(nullptr)->getCache(atoi(name));
+        string num = _msg.substr(0, 1);
+        int flag = atoi(num.c_str());
+        // 剩余内容为search的参数
+        string search = _msg.substr(2);
+        // 通过flag跳转到不同的类，返回不同的结果:长度+内容
+        // 返回Result=flag+搜索结果
+        string Result;
+        if (1 == flag) {
+            auto val = _currCache.get(search);
+            logInfo("通过%s号线程查询缓存",name);
+            if (!val.empty()) {
+                logInfo("关键词缓存已有数据,%s号线程命中缓存",name);
+                Result = val;
+            } else {
+                logInfo("推荐关键词%s为第一次查询",search);
+                KeyRecommander Key(search, _con, _cndcit, _engdict);
+                Key.doQuery();
+                _result = Key.getResult();
+                Result.append("1").append(" ").append(_result);
+                _currCache.put(search,Result);
+            }
+        } else {
+            auto val = _redis.get(_msg);
+            if (val) {
+                logInfo("网页缓存已有数据,%s号线程命中缓存",search);
+                Result = *val;
+            } else {
+                logInfo("网页关键词%s为第一次查询",search);
+                WebPageSearcher page(search);
+                _result = page.doQuery(search);
+                Result.append("2").append(" ").append(_result);
+                _redis.set(_msg, Result);
+            }
+        }
```

Comment:
we should not use if else too much to make the code hard to read.
you could split the 2 part into funcitons.
Or a much more elegant one is just `return` after line 53.
something like below:
```
if (conditon) {
...
return;
}
// do other thing;
```